src/Test/Utilities/Portable/Compilation/CompilationExtensions.cs

```diff
@@ -309,6 +309,21 @@ public static void ValidateIOperations(Func<Compilation> createCompilation)
                             Assert.True(argument.Descendants().All(n => n.IsImplicit), $"Explicit node in default argument value ({argument.Syntax.RawKind}): {argument.Syntax.ToString()}");
                         }
                     }
+
+                    // Make sure that all static member references or invocations of static methods do not have implicit IInstanceReferenceOperations
+                    // as their receivers
+                    if (operation is IMemberReferenceOperation memberReference &&
+                        memberReference.Member.IsStatic &&
+                        memberReference.Instance is IInstanceReferenceOperation)
+                    {
+                        Assert.False(memberReference.Instance.IsImplicit, $"Implicit IInstanceReceiver on {operation.Syntax}");
+                    }
+                    else if (operation is IInvocationOperation invocation &&
```

Comment:
>IInvocationOperation [](start = 42, length = 20)

It looks like IInvocationOperation should implement IMemberReferenceOperation, can be addressed separately.

Reply:
Filed [https://github.com/dotnet/roslyn/issues/23084](https://github.com/dotnet/roslyn/issues/23084).

---
In reply to: [149580815](https://github.com/dotnet/roslyn/pull/22972#discussion_r149580815) [](ancestors = 149580815)